docs/en/examples/elementalarea_with_dataobject.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Using ElementalArea with DataObject
+summary: Learn how to add ElementalArea to your models and manage it in a ModalAdmin
+---
+
+# Using ElementalArea with DataObject
+
+## Creating Model and adding ElementalArea
+
+In this example the class BlogPost will be created, it will extend the DataObject class. Also a new BlogPost Admin section will be created in the Admin panel where CMS users can manage the BlogPosts.
+
+Let's look at an example:
+
+**app/src/Admins/BlogPostsAdmin.php**
+
+```php
+<?php
+
+namespace App\Admins;
+
+use App\Models\BlogPost;
+use SilverStripe\Admin\ModelAdmin;
+
+class BlogPostsAdmin extends ModelAdmin
+{
+
+    private static string $url_segment = 'blog-posts-admin';
+
+    private static string $menu_title = 'Blog Posts';
+
+    private static string $menu_icon_class = 'font-icon-block-banner';
+
+    private static array $managed_models = [
+        BlogPost::class,
+    ];
+}
+```
+
+`$managed_models` - this static variable connects BlogPostAdmin with BlogPost, which will be created in the following code.
+When you create a DataObject you should define the connection between your DataObject and ElementalArea by adding `$has_one` relationship.
+In this example BlogPost will have one ElementalArea.
+
+**app/src/Models/BlogPost.php**
+
+```php
+<?php
+
+namespace App\Models;
+
+use App\Admins\BlogPostsAdmin;
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class BlogPost extends DataObject
+{
+    private static string $table_name = 'BlogPost';
+
+    private static $extensions = [
+        Versioned::class,
+    ];
+
+    private static array $db = [
+        'Title' => 'Varchar(255)',
+        'Content' => 'HTMLText',
+    ];
+
+    private static $has_one = [
+        'ElementalArea' => ElementalArea::class,
+    ];
+
+    private static $owns = ['ElementalArea'];
+
+    // ...
+}
+ ```
+
+If you are using ElementalArea together with DataObject, it is important to define a few following methods in the class.
+First of all, the CMSEditLink() method should be defined to generate a link for opening the editing section of the particular BlogPost.
+BaseElement uses a link, that was created in CMSEditLink() method in BlogPost class, to generate new links for opening the editing section of the particular Element.
```

Comment:
```suggestion
If you are using `ElementalArea` together with `DataObject`, there are some important methods to implement on the class.
First of all, the `CMSEditLink()` method should be defined to generate a link for opening the editing section of the particular `BlogPost`.
`BaseElement` uses a link, that was created in `CMSEditLink()` method in the `BlogPost` class, to generate new links for opening the editing section of the particular element.

Note that for nested `GridField`s this method can get more complicated. Similarly, if your class is used in multiple admins, you will have to choose one to be the canonical admin section for the purposes of this method. This example only shows the simplest case, where the `BlogPost` class is used directly in `BlogPostsAdmin`.
```

Reply:
Why is `CMSEditLink()` needed? If this is just for preview purposes, we can leave it out of this documentation and just refer to the new preview documentation.
I think there's a report that uses it, right? If so then yeah definitely leave it in but can you please just confirm that for me?

Reply:
No, user should define CMSEditLink() not for preview purpose.

Reply:
Sweet, that's fine then 👍